echopop/live/live_data_processing.py

```diff
@@ -0,0 +1,253 @@
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+from .live_biology import summarize_strata
+from .live_spatial_methods import update_population_grid
+from .sql_methods import query_dataset, sql_group_update
+
+
+def get_average_strata_weights(db_file: str, data_dict: dict, unique_columns: list):
+
+    # Get corresponding `weight_fitted_df` from the database
+    weight_fitted_sql_df = query_dataset(
+        db_file,
+        data_dict,
+        table_name="weight_stratum_df",
+        data_columns=unique_columns + ["average_weight"],
+        unique_columns=unique_columns,
+        constraint="sex == 'all'",
+    )
+    # ---- Use SQL table data if present
+    if weight_fitted_sql_df is not None and not weight_fitted_sql_df.empty:
+        # ---- Return output
+        return weight_fitted_sql_df
+    else:
+        return None
+
+
+def configure_database_paths(file_configuration: dict):
+
+    # Extract input directory settings
+    file_settings = file_configuration["input_directories"]
+
+    # Get database directory
+    database_dir = file_configuration["database_directory"]
+
+    # Update configuration
+    file_configuration["database"].update(
+        {
+            dataset: "/".join([database_dir, file_settings[dataset]["database_name"]])
+            for dataset in file_settings.keys()
+            if "database_name" in file_settings[dataset]
+        }
+    )
+
+
+def acoustic_pipeline(
+    acoustic_dict: dict,
+    strata_df: pd.DataFrame,
+    file_configuration: dict,
+    verbose: bool,
+    contrast_columns: List[str] = [],
+):
+
+    # Get spatial column
+    spatial_column = file_configuration["spatial_column"]
+    gridding_column = file_configuration["gridding_column"]
+    unique_columns = gridding_column + contrast_columns
+
+    # Get database file
+    acoustic_db = file_configuration["database"]["acoustics"]
+
+    # Get biology database file
+    biology_db = file_configuration["database"]["biology"]
+
+    # Check whether data dictionary is empty
+    if acoustic_dict["nasc_df"] is None or acoustic_dict["nasc_df"].empty:
+        # ---- Print, if verbose
+        if verbose:
+            print("No new processed acoustic data available for processing.")
+    else:
+        # Get related acoustic data
+        acoustic_df = get_nasc_sql_data(acoustic_db, acoustic_dict, unique_columns=unique_columns)
```

Comment:
This only difference in passing `acoustic_dict` and `biology_dict` here (`acoustic_pipeline`) vs in `biology_pipeline` is goes all the way down to in `query_dataset` the `get_unique_identifiers` uses different input `data_dict`. What does `get_unique_identifiers` do? There's a comment
```python
# Get unique values of each contrast column across the biological datasets
```
in that function but perhaps it is not just biological datasets?

Reply:
Got it now after verbal explanation! :)